documentation/CAMARA API Design Guide.md

```diff
@@ -0,0 +1,1120 @@
+
+**Examples**:
+- <u>Equals</u>: to search devices with a particular operating system and version or type:
+  - `GET /device?os=ios&version=17.0.1`
+  - `GET /device?type=apple,android`
+    - Search for several values separating them by "`,`".
+- <u>Inclusion</u>: if we already have a filter that searches for "equal" and we want to provide it with the possibility of searching for "inclusion", we must include the character "~"
+  - `GET /device?version=17.0.1`
+    - Search for the exact version "17.0.1"
+  - `GET /device?version=~17.0`
+    - Look for version strings that include "17.0"
+- <u>Greater than / less than</u>: new attributes need to be created with the suffixes `.(gte|gt|lte|lt)$` and included in `get` operation :
+
+```yaml
+paths:
+  /users:
+    get:  
+      parameters:
+        - $ref: "#/components/parameters/StartCreationDate"
+        - $ref: "#/components/parameters/AfterCreationDate"
+        - $ref: "#/components/parameters/EndCreationDate"
+        - $ref: "#/components/parameters/BeforeCreationDate"
+    #...
+components:
+  parameters: 
+    StartCreationDate:   #<-- component name
+      in: query
+      name: creationDate.gte    <-- query attribute for "greater - equal to" comparison 
+      required: false
+      schema:
+        format: date-time
+        type: string
+    AfterCreationDate:
+      in: query
+      name: creationDate.gt
+      required: false
+      schema:
+        format: date-time
+        type: string
+    EndCreationDate:
+      in: query
+      name: creationDate.lte
+      required: false
+      schema:
+        format: date-time
+        type: string
+    BeforeCreationDate:
+      in: query
+      name: creationDate.lt
+      required: false
+      schema:
+        format: date-time
+        type: string
+```
+
+Then the parameters can be included in the query:
+  - `GET /users?creationDate.gte=2021-01-01T00:00:00`
+    - Find users with creationDate starting from 2021
+  - `GET /users?creationDate.lt=2022-01-01T00:00:00`
+    - Find users with creationDate before 2022
+  - `GET /users?creationDate.gte=2020-01-01T00:00:00&creationDate.lte=2021-12-31T23:59:59`
+    - Search for users created between 2020 and 2021
+
+## Security
+
+<font size="3"><span style="color: blue"> Good practices to secure REST APIs </span></font>
+
+The following points can serve as a checklist to design the security mechanism of the REST APIs.
+
+1. **Simple Management**. 
+   Securing only the APIs that need to be secure. Any time the more complex solution is made "unnecessarily", it is also likely to leave a hole. 
+2. **HTTPs must be always used**. 
+   TLS ensures the confidentiality of the transported data and that the server's hostname matches the server's SSL certificate.
+   - If HTTP 2 is used to improve performance, you can even send multiple requests over a single connection, this way you will avoid the complete overhead of TCP and SSL on later requests.

```

Comment:
Note, usage of HTTP1.1 persistent connections allows also sending multiple requests over the same TCP connection.